app/models/project.rb

```diff
@@ -1,10 +1,14 @@
 class Project < ActiveRecord::Base
+  before_create :set_github_repo
   belongs_to :organization
   acts_as_ordered_taggable
   acts_as_ordered_taggable_on :technologies, :causes
 
-  attr_accessible :organization_id, :name, :url, :github_repo, :description, :notes, :cause_list, :technology_list, :is_active, :is_approved
-  validates_presence_of :name, :github_repo
+  attr_accessor :submitted_github_url
+  attr_accessible :organization_id, :name, :url, :github_repo, :description, :notes, :cause_list, :technology_list, :is_active, :is_approved, :submitted_github_url
+  validates_presence_of :name
+  validates_presence_of :github_repo, on: :update
+  validates :submitted_github_url, presence: true, format: { with: /\Ahttps?:\/\/github.com\/[\w-]+\/[\w\.-]+(\/)?\Z/i, message: 'Please enter a valid GitHub URL.' }, on: :create
```

Comment:
Line is too long. [178/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.